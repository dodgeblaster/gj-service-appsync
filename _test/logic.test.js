const logic = require('../src/logic')
describe('deploy', () => {
    test('works', async () => {
        const PROJECT_ROOT = (__dirname.slice(0, -5)) + 'exampleProject'
        const SRC_LOCATION = '/src'    
        const result = await logic(PROJECT_ROOT, SRC_LOCATION)
  

        expect(result).toMatchSnapshot()
        
    }, 300000)
})