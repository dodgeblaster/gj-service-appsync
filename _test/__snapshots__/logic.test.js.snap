// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deploy works 1`] = `
Object {
  "appsync": Object {
    "auth": Object {
      "existingKey": null,
      "state": "CREATE",
    },
    "datasourceIamRoles": Object {
      "create": Array [
        Object {
          "name": "MYNEWAPI5-datasource-LAMBDA-role",
          "state": "",
          "type": "AWS_LAMBDA",
        },
        Object {
          "name": "MYNEWAPI5datasourcedbnotes-role",
          "state": "",
          "type": "AMAZON_DYNAMODB",
        },
        Object {
          "name": "MYNEWAPI5datasourcedbexample-role",
          "state": "",
          "type": "AMAZON_DYNAMODB",
        },
      ],
    },
    "datasources": Object {
      "create": Array [
        Object {
          "lambdaArn": "arn:aws:lambda:us-east-2:251256923172:function:MYNEWAPI5-lambda",
          "name": "MYNEWAPI5datasourceLAMBDA",
          "roleArn": "arn:aws:iam::251256923172:role/MYNEWAPI5-datasource-LAMBDA-role",
          "roleName": "MYNEWAPI5-datasource-LAMBDA-role",
          "type": "AWS_LAMBDA",
        },
        Object {
          "name": "MYNEWAPI5datasourcedbnotes",
          "roleArn": "arn:aws:iam::251256923172:role/MYNEWAPI5datasourcedbnotes-role",
          "roleName": "MYNEWAPI5datasourcedbnotes-role",
          "tableName": "MYNEWAPI5notes",
          "type": "AMAZON_DYNAMODB",
        },
        Object {
          "name": "MYNEWAPI5datasourcedbexample",
          "roleArn": "arn:aws:iam::251256923172:role/MYNEWAPI5datasourcedbexample-role",
          "roleName": "MYNEWAPI5datasourcedbexample-role",
          "tableName": "MYNEWAPI5example",
          "type": "AMAZON_DYNAMODB",
        },
      ],
    },
    "graphQLApi": Object {
      "rootResource": "CREATE",
    },
    "name": "MY NEW API5",
    "resolvers": Object {
      "create": Array [
        Object {
          "datasource": "MYNEWAPI5datasourceLAMBDA",
          "field": "getHello",
          "resolverType": "AWS_LAMBDA",
          "type": "Query",
          "vtl": Object {
            "request": "{ 
            \\"version\\": \\"2017-02-28\\",
            \\"operation\\": \\"Invoke\\",
            \\"payload\\": $util.toJson($ctx)
        }",
            "response": "$util.toJson($ctx.result)",
          },
        },
        Object {
          "datasource": "MYNEWAPI5datasourceLAMBDA",
          "field": "something",
          "resolverType": "AWS_LAMBDA",
          "type": "Query",
          "vtl": Object {
            "request": "{ 
            \\"version\\": \\"2017-02-28\\",
            \\"operation\\": \\"Invoke\\",
            \\"payload\\": $util.toJson($ctx)
        }",
            "response": "$util.toJson($ctx.result)",
          },
        },
        Object {
          "datasource": "MYNEWAPI5datasourceLAMBDA",
          "field": "product",
          "resolverType": "AWS_LAMBDA",
          "type": "Query",
          "vtl": Object {
            "request": "{ 
            \\"version\\": \\"2017-02-28\\",
            \\"operation\\": \\"Invoke\\",
            \\"payload\\": $util.toJson($ctx)
        }",
            "response": "$util.toJson($ctx.result)",
          },
        },
        Object {
          "datasource": "MYNEWAPI5datasourcedbnotes",
          "field": "getOrgsNotes",
          "resolverType": "AMAZON_DYNAMODB",
          "type": "Query",
          "vtl": Object {
            "request": "{
                    \\"version\\": \\"2017-02-28\\",
                    \\"operation\\": \\"Query\\",
                    \\"key\\": {
                        \\"PK\\": $util.dynamodb.toDynamoDBJson($ctx.args.org),
                        
                        
                        \\"SK\\": \\"note\\"
                       
                    \\"filter\\": #if($context.args.filter) $util.transform.toDynamoDBFilterExpression($ctx.args.filter) #else null #end,
                    \\"limit\\": $util.defaultIfNull($ctx.args.limit, 20),
                    \\"nextToken\\": $util.toJson($util.defaultIfNullOrEmpty($ctx.args.nextToken, null)),
                    \\"condition\\": {
                        \\"expression\\": \\"PK = :pk AND begins_with(SK, :sk)\\",
                        \\"expressionNames\\": {
                            \\":pk\\": \\"PK\\",
                            \\":sk\\": \\"SK\\"
                        }
                        
                        
   
   
                    },
                }",
            "response": "$util.toJson($ctx.result)",
          },
        },
        Object {
          "datasource": "MYNEWAPI5datasourcedbexample",
          "field": "getProduct",
          "resolverType": "AMAZON_DYNAMODB",
          "type": "Query",
          "vtl": Object {
            "request": "{
                    \\"version\\": \\"2017-02-28\\",
                    \\"operation\\": \\"GetItem\\",
                    \\"key\\": {
                        \\"PK\\": $util.dynamodb.toDynamoDBJson($ctx.args.id)
                    },
                }",
            "response": "$util.toJson($ctx.result)",
          },
        },
        Object {
          "datasource": "MYNEWAPI5datasourcedbexample",
          "field": "createProduct",
          "resolverType": "AMAZON_DYNAMODB",
          "type": "Mutation",
          "vtl": Object {
            "request": "{
                    \\"version\\": \\"2017-02-28\\",
                    \\"operation\\": \\"PutItem\\",
                    \\"key\\": {
                        \\"PK\\": $util.dynamodb.toDynamoDBJson($ctx.args.input.id)
                    },
                    #set ( $ctx.args.input.createdAt = \\"$util.time.nowISO8601()\\" )
                    \\"attributeValues\\": $util.dynamodb.toMapValuesJson($ctx.args.input),
                    \\"condition\\": {
                        \\"expression\\": \\"attribute_not_exists(#PK)\\",
                        \\"expressionNames\\": {
                        \\"#PK\\": \\"PK\\"
                        },
                    },
                }",
            "response": "$util.toJson($ctx.result)",
          },
        },
        Object {
          "datasource": "MYNEWAPI5datasourcedbexample",
          "field": "removeProduct",
          "resolverType": "AMAZON_DYNAMODB",
          "type": "Mutation",
          "vtl": Object {
            "request": "{
                    \\"version\\": \\"2017-02-28\\",
                    \\"operation\\": \\"DeleteItem\\",
                    \\"key\\": {
                        \\"PK\\": $util.dynamodb.toDynamoDBJson($ctx.args.input.id)
                    },
                }",
            "response": "$util.toJson($ctx.result)",
          },
        },
      ],
    },
    "schema": Object {
      "defintion": "type Hello {
    hello: String
    id: String
    name: String
}

type Product {
    id: String
    name: String
}

type Extra {
    id: String
    product: String
}

type Coffee {
    id: String
    name: String
}

type Query {
    getHello: Hello
    something: Product
    product: Extra
    getProduct(id: String): Coffee
}


input CreateProductInput {
    id: String
    name: String
}

input RemoveProductInput {
    id: String
}

type Mutation {
    createProduct(input: CreateProductInput): Coffee
    removeProduct(input: RemoveProductInput): Coffee
}


",
      "state": "CREATE",
    },
  },
  "dynamodb": Object {
    "config": Array [
      Object {
        "GSI1": "GSI1",
        "PK": "PK",
        "SK": "SK",
        "tableName": "MYNEWAPI5notes",
      },
      Object {
        "PK": "id",
        "tableName": "MYNEWAPI5example",
      },
    ],
  },
  "monolambda": Object {
    "iamRole": "CREATE",
    "lambdaCode": "SKIP",
    "lambdaConfig": "SKIP",
  },
  "projectInfo": Object {
    "accountId": "251256923172",
    "id": "",
    "lambdaDatasourceRole": "",
    "lambdaRole": "",
    "name": "MY NEW API5",
    "projectRoot": "/Users/garyjennings/Desktop/WEBDEV/2020/NPM/gj-service-appsync/exampleProject",
    "region": "us-east-2",
    "schema": "type Hello {
    hello: String
    id: String
    name: String
}

type Product {
    id: String
    name: String
}

type Extra {
    id: String
    product: String
}

type Coffee {
    id: String
    name: String
}

type Query {
    getHello: Hello
    something: Product
    product: Extra
    getProduct(id: String): Coffee
}


input CreateProductInput {
    id: String
    name: String
}

input RemoveProductInput {
    id: String
}

type Mutation {
    createProduct(input: CreateProductInput): Coffee
    removeProduct(input: RemoveProductInput): Coffee
}


",
    "schemaChecksum": "",
    "srcLocation": "/src",
  },
}
`;
